Combobox.js

```diff
@@ -9,13 +9,12 @@ define([
 	"delite/HasDropDown",
 	"delite/keys",
 	"./list/List",
-	"./LinearLayout",
-	"./Button",
+	"./Combobox/ComboPopup",
```

Comment:
Maybe. On the other side, it somehow makes the scope of this module clearer, including when looking just at the file name. And we already have redundantly Combobox/Combobox.html (the template, which could be called generaically template.html). 

All in one:
* I'm willing to change the name, if there's agreement this would be better.
* Personally, I think both could work, while having a small preference on keeping ComboPopup rather than just Popup.
* Since you seem to hesitate on it too, I'm not sure what to do. For now, I wait for further feedback.